src/Courses/CourseDetailsTabs/ActiveTopics.tsx

```diff
@@ -11,42 +11,21 @@ interface ActiveTopicsProps {
 }
 
 export const ActiveTopics: React.FC<ActiveTopicsProps> = ({course}) => {
-    const [activeTopics, setActiveTopics] = useState<Array<TopicObject>>([]);
     const userType: UserRole = getUserRole();
     const userId: number = getUserId();
 
-    useEffect(()=>{
-        (async () => {
-            const res = await AxiosRequest.get(`/courses/topics?isOpen=true&courseId=${course.id}`);
-            let activeTopics = res.data?.data?.map((topic: any) => new TopicObject(topic));
-
-            // TODO: Currently, the backend returns topics that have extensions for any users.
-            // This filters out those topics and hides them.
-            if (userType === UserRole.STUDENT) {
-                activeTopics = activeTopics.filter((topic: TopicObject) => {
-                    // If this actually is an active topic, keep it.
-                    if (moment().isBetween(topic.startDate, topic.deadDate, 'day', '[]')) {
-                        return true;
-                    } else {
-                        return _.find(topic.studentTopicOverride, ['userId', userId]) !== undefined;
-                    }
-                }).map((topic: TopicObject) => {
-                    const override = _.find(topic.studentTopicOverride, ['userId', userId]);
-                    if (!_.isNil(override)) {
-                        delete override.id;
-                        _.assign(topic, override);
-                    }
-                    return topic;
-                });
-            }
-
-            setActiveTopics(activeTopics);
-        })();
-    }, [course.id]);
+    const activeTopics = course.units.reduce<Array<TopicObject>>((accum, unit) => {
```

Comment:
I wonder if you could have done this more simply with lodash flatten method, either way looks good

Reply:
Should we `useMemo` here?

Reply:
> I wonder if you could have done this more simply with lodash flatten method, either way looks good
I guess flatten doesn't make sense since we are also filtering

Reply:
We'd have to add every topic to the dependency array, and I'm not sure if we can do that dynamically.

I see this as more of a holdover until we add SWR.